enhancements/security/tls-artifacts-registry.md

```diff
@@ -0,0 +1,188 @@
+TLS artifacts. Both engineering and customers have expressed concern 
+about a lack of understanding what is the purpose of a particular 
+certificate or CA bundle, who issued it, and which team is responsible 
+for its contents and lifecycle. This enhancement describes a registry 
+to store necessary information about all certificates in the cluster 
+and sets requirements for TLS artifacts metadata.
+
+## Motivation
+
+TLS artifacts registry is useful for several teams in Red Hat (i.e. 
+which component manages a particular certificate) and customers - 
+for instance, helping them to separate platform-managed certificates 
+from user-managed.
+
+### User Stories
+
+- As an Openshift administrator, I want to differentiate 
+  platform-managed TLS artifacts from user-managed.
+- As an Openshift developer, I want to be able to find which team 
+  manages a particular certificate.
+- As a member of OpenShift concerned with the release process 
+  (TRT, dev, staff engineer, maybe even PM), I want to make sure all 
+  managed certificates have sufficient metadata to find the team 
+  responsible for its lifecycle.
+
+### Goals
+
+Create a registry of all managed TLS artifacts, and ensure that no new 
+certificates are added without sufficient metadata.
+Add a test to the conformance suite that verifies the actual cluster TLS 
+artifacts have necessary metadata.
+
+### Non-Goals
+
+## Proposal
+
+* Define a structure for the certificate registry origin repo
+  * Each TLS artifact can be either
+    * on-disk file
+    * in-cluster secret or config map
+  * Additional expected metadata stored for this artifact
+    * JIRA component
+    * TLS artifact description (one sentence describing the purpose 
+      of this artifact)
+  * Each TLS artifact is associated with a list of owners, responsible 
+    for the lifecycle of this artifact
+* Create a new test, which ensures that all secrets are present in the 
+  registry
+  * Some secrets may be versioned, contain hash or IP address / DNS 
+    address - the test should strip this information and match it to 
+    the canonical location (i.e. 
+    `etcd-peer-ip-10-10-10-10.us-east-1.internal` becomes 
+    `etcd-peer-for-master-2` etc.)
+* Create a test that verifies that TLS artifacts in a cluster have 
+  necessary annotations to provide required metadata.
+* TLS artifact registry is being composed as a JSON file from several 
+  parts (usually one file per a set of owners).
+* In the `origin` repo create `hack/update-tls-ownership.json` and 
+  `hack/verify-tls-ownership.json` scripts to compose TLS artifact 
+  registry JSON from several JSONs and verify that all necessary 
+  metadata is set respectively.
+* Update library-go functions to set metadata when the TLS artifact is 
+  being generated, so that most operators would use the recommended 
+  method.
+
+### Workflow Description
+
+The cluster administrator opens the TLS artifacts registry in the `origin` repo 
```

Comment:
I know we have a test designed to detect issues with certificates and I expect folks will look at the test to see what certificates needs to be fixed or issues need to be addressed. However, I wonder if we can generate a text-based output that will summarize all of failures that admin/dev can see quickly instead of having to go through the whole markdown that contains both good and bad certs.

Reply:
The test will exclude known violations. Currently all certs are added there, so the test would fail only on the certificates added in the PR